src/components/scene/pool.js

```diff
@@ -0,0 +1,103 @@
+var debug = require('../../utils/debug');
+var register = require('../../core/component').registerComponent;
+
+var warn = debug('components:pool:warn');
+
+/**
+ * Pool component.
+ * It initialized a pool of entities that will be reused.
+ * It avoids creating and destroying the same kind entities
+ * in dynamic scenes. Useful for example in a game where you want
+ * to reuse enemies entities
+ *
+ * @member {array} poolEls - Available entities in the pool.
+ * @member {array} pooledEls - Entities of the pool in use.
+ *
+ */
+module.exports.Component = register('pool', {
```

Comment:
`Entity pool` is more clear than just `pool` but due its nature, we'll have plenty of instances on a-scene therefore we'll have `entity-pool__whatever` as @dmarcos said and it looks quite ugly, so I would choose to use just `pool`